src/compiler/checker.ts

```diff
@@ -4196,9 +4317,10 @@ namespace ts {
         }
 
         // Get type from reference to class or interface
-        function getTypeFromClassOrInterfaceReference(node: TypeReferenceNode | ExpressionWithTypeArguments, symbol: Symbol): Type {
+        function getTypeFromClassOrInterfaceReference(node: TypeReferenceNode | ExpressionWithTypeArguments | JSDocTypeReference, symbol: Symbol): Type {

```

Comment:
Nathan and I tried for a while and couldn't come up with a good name that wasn't taken. Open to suggestions.